pkg/csi/service/vanilla/controller.go

```diff
@@ -906,6 +921,464 @@ func (c *controller) getAccessibleTopologiesForDatastore(ctx context.Context, vc
+				},
+			}
+			volTaskAlreadyRegistered = true
+		} else if volumeOperationDetails.OperationDetails.TaskStatus ==
+			cnsvolumeoperationrequest.TaskInvocationStatusInProgress &&
+			volumeOperationDetails.OperationDetails.TaskID != "" {
+			// If task is already created in CNS for this volume but task is in progress,
+			// we need to monitor the task to check if volume creation is complete or not.
+			log.Infof("Volume with name %s has CreateVolume task %s pending on VC %q.",
+				req.Name, volumeOperationDetails.OperationDetails.TaskID,
+				volumeOperationDetails.OperationDetails.VCenterServer)
+
+			// Get vCenter instance.
+			vcHost = volumeOperationDetails.OperationDetails.VCenterServer
+			vcenter, err = common.GetVCenterFromVCHost(ctx, c.managers.VcenterManager, vcHost)
+			if err != nil {
+				return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
+					"failed to fetch vCenter instance %q. Error: %+v", vcHost, err)
+			}
+			// Create task object.
+			taskMoRef := types.ManagedObjectReference{
+				Type:  "Task",
+				Value: volumeOperationDetails.OperationDetails.TaskID,
+			}
+			task := object.NewTask(vcenter.Client.Client, taskMoRef)
+
+			defer func() {
+				// Persist the operation details before returning. Only success or error
+				// needs to be stored as InProgress details are stored when the task is
+				// created on CNS.
+				if volumeOperationDetails != nil && volumeOperationDetails.OperationDetails != nil &&
+					volumeOperationDetails.OperationDetails.TaskStatus !=
+						cnsvolumeoperationrequest.TaskInvocationStatusInProgress {
+					err := operationStore.StoreRequestDetails(ctx, volumeOperationDetails)
+					if err != nil {
+						log.Warnf("failed to store CreateVolume operation request details with error: %v", err)
+					}
+				}
+			}()
+
+			// Monitor CreateVolume task.
+			volumeMgr, err = common.GetVolumeManagerFromVCHost(ctx, c.managers, vcHost)
+			if err != nil {
+				return nil, csifault.CSIInternalFault, logger.LogNewErrorCode(log, codes.Internal, err.Error())
+			}
+			volumeInfo, faultType, err = volumeMgr.MonitorCreateVolumeTask(ctx,
+				&volumeOperationDetails, task, req.Name, c.managers.CnsConfig.Global.ClusterID)
+			if err != nil {
+				return nil, faultType, logger.LogNewErrorCodef(log, codes.Internal,
+					"failed to monitor task for volume %s on VC %q. Error: %+v", req.Name, vcHost, err)
+			}
+			volTaskAlreadyRegistered = true
+		}
+	}
+
+	var (
+		sharedDatastores    []*cnsvsphere.DatastoreInfo
+		topologyRequirement *csi.TopologyRequirement
+	)
+	// Get accessibility requirements.
+	topologyRequirement = req.GetAccessibilityRequirements()
+	if topologyRequirement == nil {
+		return nil, csifault.CSIInvalidArgumentFault, logger.LogNewErrorCode(log, codes.InvalidArgument,
+			"accessibility requirements cannot be nil for a multi-VC environment")
+	}
+	// Club the accessibility requirements according to the VC they belong to.
```

Comment:
Club the accessibility requirements => get the accessibility requirements

Reply:
Done